src/arch/x86/x86_64/src/bios.rs

```diff
@@ -0,0 +1,380 @@
+/* SPDX-License-Identifier: GPL-2.0-only */
+
+// from github.com:akaros/vmm-akaros/xhype/xhype
+use crate::consts::x86::*;
+use crate::utils::round_up_4k;
+use core::fmt::Write;
+use core::mem::size_of;
+use core::ptr;
+use print;
+// https://uefi.org/sites/default/files/resources/ACPI_6_3_May16.pdf
+
+// Table 5-27 RSDP Structure
+#[repr(packed)]
+#[derive(Default)]
+pub struct AcpiTableRsdp {
+    pub signature: [u8; 8],         /* ACPI signature, contains "RSD PTR " */
+    pub checksum: u8,               /* ACPI 1.0 checksum */
+    pub oem_id: [u8; 6],            /* OEM identification */
+    pub revision: u8,               /* Must be (0) for ACPI 1.0 or (2) for ACPI 2.0+ */
+    pub rsdt_physical_address: u32, /* 32-bit physical address of the RSDT */
+    pub length: u32,                /* Table length in bytes, including header (ACPI 2.0+) */
+    pub xsdt_physical_address: u64, /* 64-bit physical address of the XSDT (ACPI 2.0+) */
+    pub extended_checksum: u8,      /* Checksum of entire table (ACPI 2.0+) */
+    pub reserved: [u8; 3],          /* Reserved, must be zero */
+}
+pub const ACPI_RSDP_CHECKSUM_LENGTH: usize = 20;
+pub const ACPI_RSDP_XCHECKSUM_LENGTH: usize = 36;
+pub const ACPI_RSDP_CHECKSUM_OFFSET: usize = 8;
+pub const ACPI_RSDP_XCHECKSUM_OFFSET: usize = 32;
+
+#[repr(packed)]
+#[derive(Default)]
+pub struct AcpiTableHeader {
+    pub signature: [u8; 4],         /* ASCII table signature */
+    pub length: u32,                /* Length of table in bytes, including this header */
+    pub revision: u8,               /* ACPI Specification minor version number */
+    pub checksum: u8,               /* To make sum of entire table == 0 */
+    pub oem_id: [u8; 6],            /* ASCII OEM identification */
+    pub oem_table_id: [u8; 8],      /* ASCII OEM table identification */
+    pub oem_revision: u32,          /* OEM revision number */
+    pub asl_compiler_id: [u8; 4],   /* ASCII ASL compiler vendor ID */
+    pub asl_compiler_revision: u32, /* ASL compiler version */
+}
+pub const ACPI_TABLE_HEADER_CHECKSUM_OFFSET: usize = 9;
+
+impl AcpiTableHeader {
+    pub fn new() -> Self {
+        AcpiTableHeader { revision: 2, checksum: 0, oem_id: *b"AKAROS", oem_table_id: *b"ALPHABET", oem_revision: 0, asl_compiler_id: *b"RON ", asl_compiler_revision: 0, ..Default::default() }
```

Comment:
Does it make sense to have the oem_id as AKAROS if we're using it?

Reply:
oops, let's pick a better name. This is kind of a nod to the past.

Reply:
try that.